lms/views/basic_lti_launch.py

```diff
@@ -94,38 +93,40 @@ def store_lti_data(self):
     @view_config(canvas_file=True)
     def canvas_file_basic_lti_launch(self):
         """
-        Respond to a Canvas file assignment launch.
+        Respond to a Canvas file assignment launch which is not db_configured.
 
         Canvas file assignment launch requests have a ``file_id`` request
         parameter, which is the Canvas instance's ID for the file. To display
         the assignment we have to use this ``file_id`` to get a download URL
         for the file from the Canvas API. We then pass that download URL to
         Via. We have to re-do this file-ID-for-download-URL exchange on every
         single launch because Canvas's download URLs are temporary.
-        """
 
+        Note that this only apply to assignments configured but not yet launched
+        after canvas assignments are also DB configured see: js_config._create_assignment_api
+        """
         course_id = self.request.params["custom_canvas_course_id"]
         file_id = self.request.params["file_id"]
         resource_link_id = self.request.params["resource_link_id"]
 
-        # Normally this would be done during `configure_assignment()` but
-        # Canvas skips that step. We are doing this to ensure that there is a
-        # module item configuration. As a result of this we can rely on this
-        # being around in future code.
+        # This URL is mostly for show. We just want to ensure that a module
+        # configuration exists. If we're going to do that we might as well
+        # make sure this URL is meaningful.
+        document_url = f"canvas://file/course/{course_id}/file_id/{file_id}"
+
+        # Newly configured canvas assignments make it to the database
+        # but for exiting ones that never got launched we keep this call to persist them.
         self.assignment_service.set_document_url(
             # This URL is mostly for show. We just want to ensure that a module
             # configuration exists. If we're going to do that we might as well
             # make sure this URL is meaningful.
-            document_url=f"canvas://file/course/{course_id}/file_id/{file_id}",
+            document_url=document_url,
             tool_consumer_instance_guid=self.request.params[
                 "tool_consumer_instance_guid"
             ],
             resource_link_id=resource_link_id,
         )
-
-        self.context.js_config.add_canvas_file_id(course_id, resource_link_id, file_id)
```

Comment:
I think there's a refactoring here where you've removed the call to `add_canvas_file_id()` from this view, and moved it into `JSConfig` instead (the new `elif document_url.startswith("canvas://")` case). I can see that potentially making sense because `JSConfig` already contains a case for `if document_url.startswith("blackboard://")`. But would you mind sending this as a separate PR first? I think it's a logically separate refactoring. This stuff is quite hard to wrap your head around so I think one change at a time is going to make it easier for me to review